### DISCUSSION TITLE
firefox: cannot communicate with 1Password

### DISCUSSION THREAD

**Discussion by jovial-snyder** [netblue30/firejail#5985] [Q&A]
I am running an up-to-date Firefox with Firejail 0.9.72 and the firefox.profile. I am also using 1Password (installed via the deb, and it's running without Firejail, so no sandbox at all). My goal now is to have the [1Password Firefox addon](https://addons.mozilla.org/en-US/firefox/addon/1password-x-password-manager/?src=search) automatically unlocked when the 1Password desktop is unlocked. This works (as expected) when I run Firefox without Firejail, but doesn't when I use Firejail.

So the situation is very similar to using [Firefox with the KeePassXC addon](https://github.com/netblue30/firejail/discussions/5444). But so far, my attempts to make it work have been unsuccessful. I think this could be of interest to many people, seeing that 1PW is a widely used password manager, especially on company-owned devices.

Any ideas?

Relates to:

* #4874

**Comment by ghost** [comment]
Can you collect the `${RUNUSER}` paths 1PW uses when you run Firefox without Firejail and post that here? If it is indeed similar to https://github.com/netblue30/firejail/discussions/5444 you will need to integrate those locations as indicated in that thread.

**Comment by jovial-snyder** [comment]
Hey. I wouldn't even know where to begin. No idea what the env variable `RUNUSER` is supposed to be or how I would get it from a running application like 1PW. As far as I can tell it's not explained in the Firejail documentation. The only thing I am aware of is https://man7.org/linux/man-pages/man1/runuser.1.html but I have no clue how this relates to Firejail or how I would get the paths I am looking for. It's not clear to me how they even found those KeePassXC paths in the issue you mentioned.

**Reply by ghost** [reply]
${RUNUSER} is a Firejail construct, comparable to ${DESKTOP}, ${DOCUMENTS}, ${DOWNLOADS} etc from the XDG specification. It expands to /run/user/\<uid\> so if your user is 1000 (a default on most Linux distro's) you will see a set of dirs/files exposed under `/run/user/1000`.

I've tried 1PW for a while to assist debugging this and - depending on configuration - I noticed it creates:

- /run/user/1000/1Password-BrowserSupport.sock
- /run/user/1000/s.sock <-- this only gets exposed if you enabled SSH Agent in Preferences (under devtools IIRC), which implies allowing access to ${HOME}/.ssh and ${HOME}/.1password

The latest 1PW version at least can also use `polkit` apparently. For that a `whitelist /usr/share/polkit-1/actions/com.1password.1Password.policy` is needed in `1password.profile`.
And it exposes `com.onepassword.OnePassword` on the D-Bus session bus so we could filter on that too. BUT this is all relevant to firejailing 1PW and - sadly enough - doesn't help in your quest to communicate with Firefox as far as I have been able to test. Someone more familiar with 1PW might have ideas how this extra information can help in this context.

I couldn't even get unsandboxed 1PW to communicate with an unsandboxed Firefox. Always needed to unlock the web extension in the browser even though 1PW is unlocked. I'm afraid I gave up after a few days. At least we can ascertain that our current 1password.profile isn't very helpfull to users and IMO should be removed from `firecfg`.

**Reply by jovial-snyder** [reply]
I don't think the 1Password profile is the problem, it's the Firefox profile. I can get KeePassXC (not sandboxed) to work with Firefox (sandboxed) if I follow the steps mentioned in my initial post. But 1Password (not sandboxed) with Firefox (sandboxed) does not work, even if I try to add those changes you mentioned.

1PW + Firefox (both not sandboxed) does work for me after applying [this](https://1password.community/discussion/comment/689837/#Comment_689837) fix.

**Comment by ghost** [comment]
> 1PW + Firefox (both not sandboxed) does work for me after applying [this](https://1password.community/discussion/comment/689837/#Comment_689837) fix.

Thanks for the info. But the [1Password package from AUR](https://aur.archlinux.org/packages/1password) I used to try to reproduce does have all the correct permissions and ownerships already. It still doesn't work for me here I'm afraid. I did notice it creates ${HOME}/.1Password (besides ${HOME}/.config/1Password). Maybe the profile needs whitelisted access to that path too, not sure. I do admit the 1Password app is a total black box to me. Hopefully someone with more insights chimes in...

**Comment by jaredmo** [comment]
I'm struggling with this as well. Anyone find a solution on the whitelist? Firefox and 1Password work together when Firejail is disabled.

**Reply by reedriley** [reply]
I spent some time debugging this on NixOS and found the root cause. Sharing findings in case it helps.

The problem: 1Password authenticates browser extension connections via SO_PEERCRED on the Unix socket (1Password-BrowserSupport.sock). It checks that the connecting process's effective GID equals the onepassword group (GID 31001). On NixOS, 1Password-BrowserSupport is a setgid binary (owner root:onepassword) installed via a security wrapper in /run/wrappers/.

This fails inside firejail because firejail mounts its tmpfs overlays with MS_NOSUID (hardcoded in src/firejail/fs.c, fs_tmpfs()). All bind-mounted paths into the sandbox inherit the nosuid flag. This means the kernel silently ignores the setgid bit on 1Password-BrowserSupport when it runs inside the sandbox — the process keeps the user's primary GID instead of switching to the onepassword group. The 1Password daemon then sees the wrong GID via SO_PEERCRED and rejects the connection:

```
  WARN  invalid group attempted to connect, rejecting remote
  ERROR Failed to accept new connection.: PipeAuthError(NoCreds)
```

What doesn't work:
  - ignore nonewprivs / ignore noroot — These only disable firejail's own PR_SET_NO_NEW_PRIVS prctl and user namespace restrictions. They do NOT affect the kernel-level nosuid mount flag on the tmpfs.
  - ignore nogroups / ignore caps.drop all — Same story; these are firejail-level restrictions, not kernel mount flags.
  - Adding the user to the onepassword supplementary group — SO_PEERCRED only returns the effective GID (struct ucred { pid, uid, gid }), not supplementary groups. So even if the user is in the group, the daemon still sees the wrong GID.
